Source/WebKit/WebProcess/WebPage/WebFrame.h

```diff
@@ -269,6 +269,7 @@ class WebFrame : public API::ObjectImpl<API::Object::Type::BundleFrame>, public
 #endif
     std::optional<NavigatingToAppBoundDomain> m_isNavigatingToAppBoundDomain;
     Markable<WebCore::LayerHostingContextIdentifier> m_layerHostingContextIdentifier;
+    std::optional<WebCore::FrameIdentifier> m_frameIDBeforeProvisionalNavigation;
```

Comment:
nit: Markable uses less memory for this persistent type.  You'd have to use asOptional() in WebFrame::commitProvisionalFrame